### PR TITLE
Add time conversion utilities.

### PR DESCRIPTION
**Pull request by Spawek**

### CODE REVIEW

File: src/session/error.rs [hunk 1]
```diff
@@ -91,6 +91,9 @@ pub enum ParseError {
     Malformed(Box<dyn std::error::Error + Send + Sync>),
     /// An error occurred when decoding bytes of a proto message.
     Decode(prost::DecodeError),
+    /// An error occurred when converting time micros from proto
+    /// to `std::time::SystemTime`.
+    TimeMicrosConversion(TimeMicrosConversionError)
```
Comment (panhania) [inline src/session/error.rs:96]: Likewise, I don't think we should add too much variants into the `ParseError`. `TmeMicrosConversion` falls under `Malformed` IMO.

File: src/session/time.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    micros: u64,
+) -> Result<std::time::SystemTime, TimeMicrosConversionError> {
+    let result = std::time::UNIX_EPOCH
+        .checked_add(std::time::Duration::from_micros(micros));
+
+    if result.is_none() {
+        return Err(TimeMicrosConversionError { micros });
+    }
+
+    Ok(result.unwrap())
```
Comment (panhania) [inline src/session/time.rs:15]: Avoid using `Option::unwrap` in any code. Here, this should be easily expressible with `Option::ok_or`.

**Review by panhania — CHANGES_REQUESTED** [review]

File: src/session/error.rs (hunk 1 above)
Comment (Spawek) [reply src/session/error.rs]: Removed the variant - merged it into `Malformed`.

File: src/session/time.rs (hunk 2 above)
Comment (Spawek) [reply src/session/time.rs]: `ok_or` makes that code much better! Thanks for the suggestion!

File: src/session/error.rs [hunk 3]
```diff
@@ -90,7 +90,7 @@ pub enum ParseError {
     /// An error occurred because the decoded proto message was malformed.
     Malformed(Box<dyn std::error::Error + Send + Sync>),
     /// An error occurred when decoding bytes of a proto message.
-    Decode(prost::DecodeError),
```
Comment (panhania) [inline src/session/error.rs:93]: Why delete this trailing comma...?

File: src/session/error.rs [hunk 4]
```diff
@@ -202,3 +202,32 @@ impl<T: Debug> std::error::Error for UnsupportedValueError<T> {
         None
     }
 }
+
+/// An error type for situations where time micros cannot be converted
+/// to `std::time::SystemTime`.
+#[derive(Debug)]
+pub struct TimeMicrosConversionError {
+    /// Time micros value causing the conversion error.
+    pub micros: u64
```
Comment (panhania) [inline src/session/error.rs:211]: Oh, and here is a missing trailing comma.

File: src/session/error.rs (hunk 3 above)
Comment (Spawek) [reply src/session/error.rs:93]: Added it back.

**Review by panhania — APPROVED** [review]

**Pull request merged**